File: CSharpRewrite/CSharpRewrite/Assignments/K01.cs
using CSharpRewrite.Shared;

namespace CSharpRewrite.Assignments;

public class K01 : IQuestion
{
    public void Question1()
    {
        Console.WriteLine("Hello World! ようこそC#言語の世界へ！");
    }

    public void Question2()
    {
        Console.WriteLine("Hello World!");
        Console.WriteLine("ようこそ");
        Console.WriteLine("C#言語の世界へ！");
    }

    public void Question3()
    {
        Console.WriteLine("整数：" + 12345);
        Console.WriteLine("実数：" + 123.456789);
        Console.WriteLine("文字：" + 'X');
        Console.WriteLine("文字列：" + "ABCdef");
    }

    public void Question4()
    {
        Console.WriteLine("              ##");
        Console.WriteLine("             #  #");
        Console.WriteLine("             #  #");
        Console.WriteLine("            #    #");
        Console.WriteLine("           #      #");
        Console.WriteLine("         ##        ##");
        Console.WriteLine("       ##            ##");
        Console.WriteLine("    ###                ###");
        Console.WriteLine(" ###       ##    ##       ###");
        Console.WriteLine("##        #  #  #  #        ##");
        Console.WriteLine("##         ##    ##         ##");
        Console.WriteLine(" ##     #            #     ##");
        Console.WriteLine("  ###     ##########     ###");
        Console.WriteLine("     ###              ###");
        Console.WriteLine("        ##############");
    }
}

File: RubyRewrite/lib/assignments/k01.rb
# frozen_string_literal: true

require_relative '../shared/question'

class K01 < Question
  def question1
    puts 'Hello World! ようこそRuby言語の世界へ！'
  end

  def question2
    puts 'Hello World!'
    puts 'ようこそ'
    puts 'Ruby言語の世界へ！'
  end

  def question3
    puts format('整数：%d', 12_345)
    puts format('実数：%f', 123.456789)
    puts format('文字：%c', 'X')
    puts format('文字列：%s', 'ABCdef')
  end

  def question4
    puts '              ##'
    puts '             #  #'
    puts '             #  #'
    puts '            #    #'
    puts '           #      #'
    puts '         ##        ##'
    puts '       ##            ##'
    puts '    ###                ###'
    puts ' ###       ##    ##       ###'
    puts '##        #  #  #  #        ##'
    puts '##         ##    ##         ##'
    puts ' ##     #            #     ##'
    puts '  ###     ##########     ###'
    puts '     ###              ###'
    puts '        ##############'
  end
end
